src/cme.jl

```diff
@@ -0,0 +1,76 @@
+
+"""
+Performs a breadth-first graph traversal of a  to enumerate all possible states.
+
+The truncation kwarg should be given as a vector of Int, denoting the maximum species' counts.
+If, upon taking traversing a reaction the amount of any species is greater than the value in the truncation vector, 
+the graph traversal will not add the state to the list of possible states.
+"""
+function traverse_reactionsystem(rs, u0; truncation=nothing)
+    num_molecules = sum(u0)
+    num_species = length(u0)
+    state_space = []
+    Q = Queue{typeof(u0)}()
+    enqueue!(Q, u0)
+    push!(state_space, u0)
+    nsm = netstoichmat(rs)
+    edges = eachcol(nsm)
+    while !isempty(Q)
+        w = dequeue!(Q)
+        for e in edges
+            x = w .+ e
+            if x ∉ state_space && all(y >= 0 for y in x)
+                if truncation === nothing 
+                    if sum(x) > num_molecules 
```

Comment:
Do we want the threshold this tight? What if we have the simple system `A <--> 2B`? If `A --> 2B` is ordered first, wouldn't we immediately violate this condition?

Reply:
Hmm, if we know a network is conservative can we give a rigorous upper bound on the amount of molecules we'll ever see? I don't know the theory here, but if there are any results we could add a method to check for conservativeness and use the related bound.